Try 50.1.0 again

AntoinePrv opened the pull request:
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Close #73
Close #72
Close #70

- Deactivated mimalloc on ppc64le since it was failing
- Move everything to LTO thin since memory was saturated on all builds (pending failures)

regro-cf-autotick-bot committed:
updated v50.1.0

xhochy committed:
Export C(XX)FLAGS

xhochy committed:
Use longcall on ppc64le

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

AntoinePrv committed:
Try thin LTO

AntoinePrv committed:
Desactivate mimalloc

AntoinePrv committed:
LTO thin on all builds

AntoinePrv commented:
@conda-forge/datafusion @xhochy This is good now.
All rebuilds can be relaunched after this is merged.

xhochy reviewed (approved):
I can merge in ~1 week if the maintainers are not responding.

xhochy committed:
Re-enable ppc64le

xhochy committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.12.10.31.43

Other tools:
- conda-build 25.11.1
- rattler-build 0.54.0
- rattler-build-conda-compat 1.4.10
